README.md

```diff
@@ -157,6 +166,19 @@ cmake .. -GNinja -DCMAKE_BUILD_TYPE=DEBUG \
   -DCMAKE_CXX_COMPILER=/usr/bin/clang++-14
 ```
 
+In MacOS, if you install llvm clang, you need to explicitly specify to use llvm clang.
+
+```shell
+# Add the following line to your shell environment, e.g. ~/.bash_profile
+# to make sure llvm first in your PATH
+export PATH="/opt/homebrew/opt/llvm/bin:$PATH"
+export CC="/opt/homebrew/opt/llvm/bin/clang"
+export CXX="/opt/homebrew/opt/llvm/bin/clang++"
+
+# or use CMAKE_C_COMPILER and CMAKE_CXX_COMPILER to specify the compiler, like this:
+cmake .. -GNinja -DCMAKE_BUILD_TYPE=DEBUG -DCMAKE_C_COMPILER=/opt/homebrew/opt/llvm/bin/clang -DCMAKE_CXX_COMPILER=/opt/homebrew/opt/llvm/bin/clang++
+```
+
```

Comment:
```suggestion
Add the following line to your shell environment, e.g. `~/.bash_profile`.
```shell
# Make sure llvm comes first in your PATH
export PATH="/opt/homebrew/opt/llvm/bin:$PATH"
export CC="/opt/homebrew/opt/llvm/bin/clang"
export CXX="/opt/homebrew/opt/llvm/bin/clang++"
```

Or use `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER` to specify the compiler. For example:
```shell
cmake .. -GNinja -DCMAKE_BUILD_TYPE=DEBUG -DCMAKE_C_COMPILER=/opt/homebrew/opt/llvm/bin/clang -DCMAKE_CXX_COMPILER=/opt/homebrew/opt/llvm/bin/clang++
```

```